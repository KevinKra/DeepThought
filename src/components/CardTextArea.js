import React from "react";
import "../styles/_CardTextArea.scss";
class CardTextArea extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }
  render() {
    return (
      <form className="cardTextArea-form">
        <label>
          {this.props.label}
          <textarea />
        </label>
        <button type="">SUBMIT</button>
      </form>
    );
  }
}

export default CardTextArea;
