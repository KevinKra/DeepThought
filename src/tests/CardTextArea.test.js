import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import CardTextArea from "../components/CardTextArea";

const defaultState = {};

describe("<CardTextArea />", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<CardTextArea />);
  });
  it("should match the snapshot", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it("should match default state", () => {
    expect(wrapper.state()).toEqual(defaultState);
  });
  // it("should invoke the handleClick method when the button is clicked", () => {
  //   wrapper.instance().props.handleClick = jest.fn();
  //   expect(wrapper.find("button").exists()).toBe(true);
  //   wrapper.find("button").simulate("click");
  //   expect(wrapper.instance().props.handleClick).toHaveBeenCalled();
    //--- handleClick lives in ReviewPage component
    // const baseProps = {
    //   this.setState({
    //     hasBeenClicked: true
    //   }),
    //   trackEvent: jest.fn(),
    // };
  // });
});
