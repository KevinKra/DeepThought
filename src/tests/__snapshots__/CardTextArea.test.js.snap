// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardTextArea /> should match the snapshot 1`] = `
<form
  className="cardTextArea-form"
>
  <label>
    <textarea />
  </label>
  <button
    type="submit"
  >
    SUBMIT
  </button>
</form>
`;
